chaosreliably/controls/experiment.py

```diff
@@ -0,0 +1,310 @@
+from typing import Any, cast
+
+from chaoslib.types import Configuration, Experiment, Secrets
+from logzero import logger
+
+from chaosreliably import get_session
+from chaosreliably.types import (
+    EntityContext,
+    EntityContextExperimentEventLabels,
+    EntityContextExperimentLabels,
+    EntityContextExperimentRunLabels,
+    EntityContextExperimentVersionLabels,
+    EntityContextMetadata,
+    EventType,
+)
+
+__all__ = ["before_experiment_control"]
+
+
+def before_experiment_control(
+    context: Experiment,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+    **kwargs: Any,
+) -> None:
+    """
+    Control run *before* the execution of an Experiment.
+
+    For a given Experiment, the control creates (if not already created) an Experiment
+    Entity Context and an Experiment Version Entity Context in the Reliably Entity
+    Server.
+
+    A unique Experiment Run Entity Context is also created, with an Experiment Event
+    Entity Context of type `EXPERIMENT_START` created, relating to the run.
+
+    The control requires the `arguments` of `commit_hash`, `source`, and `user` to be
+    provided to the control definition. If not provided, the control will simply not
+    create any Entity Contexts.
```

Comment:
Will it log somethinbg for the user to be aware of this? At least at the debug level.

Reply:
As seen, done!